### PR TITLE
Added reverse-ordered offset/multiplier and lower/upper to the parser

### PR DESCRIPTION
**Pull request by rybern**
Also added complete error messages.

Addresses #511

### CODE REVIEW

File: test/integration/bad/new/stanc.expected [hunk 1]
```diff
@@ -2042,13 +2042,13 @@ Expected "offset = " expression (not containing binary logical operators).
...
-Syntax error in 'location-scale-bad9.stan', line 1, column 27 to column 28, parsing error:
+Syntax error in 'location-scale-bad9.stan', line 1, column 28 to column 29, parsing error:
    -------------------------------------------------
      1:  data { vector<multiplier = y,> x;}
-                                     ^
+                                      ^
    -------------------------------------------------
 
-Expected ">" after "multiplier =" expression.
+Expected '>' or lower expression after multiplier expression.
```
Comment (nhuurre) [inline test/integration/bad/new/stanc.expected:2051]: I think this should be `'>' or offset expression`.

Comment (rybern) [reply test/integration/bad/new/stanc.expected]: Fixed. Nice catch!

**Comment by bob-carpenter** [comment]
Thanks.  The code looks so elegant.

I didn't see a test for the different orders of lower and upper that would match the one for different orders of offset and multiplier.

**Comment by rybern** [comment]
@bob-carpenter good point, added a small test

**Pull request merged**